internal/opts/config.go

```diff
@@ -0,0 +1,192 @@
+}
+
+func (c *Config) parseConfig() {
+	viper.Unmarshal(c)
+	//Handling time duration variable unmarshalling
+	if c.ReplPollIntervalString != "" {
+		replicationPollInterval, err := time.ParseDuration(c.ReplPollIntervalString)
+		if err != nil {
+			log.Panicf("Failed to read Replication poll iterval value from config %v", err)
+		}
+		c.ReplPollInterval = replicationPollInterval
+	}
+	//Append node name to default db folder location
+	if c.DbFolder == "" {
+		c.DbFolder = path.Join(c.RootFolder, c.NodeName, "data")
+	}
+
+	if c.NexusLogDir == "" {
+		c.NexusLogDir = path.Join(c.RootFolder, c.NodeName, "nexus")
+	}
+
+	if c.NexusSnapDir == "" {
+		c.NexusSnapDir = path.Join(c.RootFolder, c.NodeName, "snap")
+	}
+
+	c.validateFlags()
+}
+
+func (c *Config) validateFlags() {
+	if c.ListenAddr != "" && strings.IndexRune(c.ListenAddr, ':') < 0 {
+		log.Panicf("given listen address: %s is invalid, must be in host:port format", c.ListenAddr)
+	}
+	if c.StatsdAddr != "" && strings.IndexRune(c.StatsdAddr, ':') < 0 {
+		log.Panicf("given StatsD address: %s is invalid, must be in host:port format", c.StatsdAddr)
+	}
+
+	if c.DisklessMode && strings.ToLower(c.DbEngine) == "rocksdb" {
+		log.Panicf("diskless is available only on Badger storage")
+	}
+
+	if c.DbEngineIni != "" {
+		if _, err := os.Stat(c.DbEngineIni); err != nil && os.IsNotExist(err) {
+			log.Panicf("given storage configuration file: %s does not exist", c.DbEngineIni)
+		}
+	}
+
+	if c.DbRole == "slave" && c.DisableAutoMasterDisc {
+		if c.ReplicationMasterAddr == "" || strings.IndexRune(c.ReplicationMasterAddr, ':') < 0 {
+			log.Panicf("given master address: %s for replication is invalid, must be in host:port format", c.ReplicationMasterAddr)
+		}
+	}
+}
+
+func (c *Config) Print() {
+	f := reflect.TypeOf(*c)
+	v := reflect.ValueOf(*c)
+	for i := 0; i < v.NumField(); i++ {
+		value := v.Field(i).Interface()
+		tag := f.Field(i).Tag
+		name := tag.Get("mapstructure")
+		if name == "" {
+			continue
+		}
+		log.Printf("%s (%s) : %v\n", name, tag.Get("desc"), value)
+	}
+}
+
+func (c *Config) Init(cfgFile string) {
+	loadConfigFile(cfgFile)
+	applyConfigOverrides()
+	c.parseConfig()
+}
+
+func loadConfigFile(cfgFile string) {
+	if cfgFile != "" {
+		absPath, err := filepath.Abs(cfgFile)
+		if err != nil {
+			log.Panicf("Failed to convert cfg file to abs path %v", err)
+		}
+		viper.SetConfigFile(absPath)
+	} else {
+		// Search config in /etc/default directory with name "dkvsrv.yaml"
+		viper.AddConfigPath("/etc/default")
+		viper.SetConfigType("yaml")
+		viper.SetConfigName("dkvsrv")
+	}
+	viper.AutomaticEnv()
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	} else {
+		fmt.Println("No config file found")
+		flag.Usage()
+		os.Exit(2)
+	}
+}
+
+func applyConfigOverrides() {
+	flag.CommandLine.VisitUnknowns(func(f *flag.Flag) {
+		// Apply the viper config value to the flag when the flag is not set and viper has a value
```

Comment:
Couldn't find the relevant documentation for this. Can you please point to it? Also, this comments seems to be misleading. Please check.

Reply:
This is from a PR on spf12/plags which is very useful for our use-case https://github.com/spf13/pflag/pull/285